### PR TITLE
BUGFIX: some keys could not be pressed at the same time

### PR DESCRIPTION
**Pull request by rschrader**
This fixes an issue where some keys prevented other keys to be pressed
when they are pressed. To fix this, the variable used to read the GPIOs
is marked as volatile. This bug maybe only occurs with arm-none-eabi-gcc
version 9.1.0.

### CODE REVIEW

**Comment by rschrader** [comment]
Not sure if you are interested in my sanity commits as well or just in the bugfix commit. Let me know what you think. The sanity commits just made it way easier to work in my setup and might be helpful for you as well.

Could do some more sanity stuff while working on the encryption topic if your are interested.

**Comment by LucidityCrash** [comment]
That might explain the issues I had when compiling on Ubuntu 19.04 ... the key rollover got really funky, both in 6kro and nkro ... I'll need to try this :)  Like the change to the Make file so you don't need to change the code to build one hand or the other.

File: redox-w-keyboard-basic/program_left.sh [hunk 1]
```diff
@@ -1,12 +1,12 @@
 #!/bin/bash
 echo '=============================== MAKING ================================'
 cd custom/armgcc
-make
+make keyboard_side=left
```
Comment (LucidityCrash) [inline redox-w-keyboard-basic/program_left.sh]: could you not just make the keyboard side a required parameter then you wouldn't need 2 programming scripts ?

Comment (rschrader) [reply redox-w-keyboard-basic/program_left.sh]: Sure just wanted not rework everything... But in that case I would propose to remove the scripts completely and add programming targets to the make files. What do you think?

Comment (mattdibi) [reply redox-w-keyboard-basic/program_left.sh]: I think the two programming script and the main makefile are good right now. I would ask you to update the readme with the instruction to use the scripts you added instead.

**Comment by mattdibi** [comment]
Hi,

thank you for contributing. Great work!

I just tested your changes with [the recent changes](https://github.com/mattdibi/redox-w-firmware/pulls?q=is%3Apr+is%3Aclosed) and everything seems to be in working order.

As I said in the comment I would ask you to update the readme with the instructions for the new scripts. Once merged I'll update the precompiled hexes to incorporate the recent changes. :D

**Comment by rschrader** [comment]
Thanks!

Made the changes to the README. Also i added some location awareness to the scripts so that they can called from any location

**Comment by rschrader** [comment]
You should consider removing the precompiled binaries from the repository. Source repositories should not contain any binaries if possible ;). Github has a release feature for that.

**Comment by mattdibi** [comment]
Merged. Thank you!

Regarding the precompiled: I was thinking about that this morning. I started with the `precompiled` folder to follow the format used by the Mitosis repository but I think it's time to change. ;)

**Comment by mattdibi** [comment]
I was too eager to merge the PR and didn't spot the issue: the changes to the programming scripts broke the scripts. The `HEX` variable is empty and they don't write anything into the

**Comment by mattdibi** [comment]
I'm sorry I reverted back the scripts to the position dependent version. I couldn't figure out how to evaluate the readlink command correctly. If you happen to fix this open a new PR.

**Pull request merged**